frontend/src/pages/modelServing/screens/projects/NIMServiceModal/DeployNIMServiceModal.tsx

```diff
@@ -0,0 +1,327 @@
+  const isInferenceServiceNameWithinLimit =
+    translateDisplayNameForK8s(createDataInferenceService.name).length <= 253;
+
+  const acceleratorProfileState = useServingAcceleratorProfile(
+    editInfo?.servingRuntimeEditInfo?.servingRuntime,
+    editInfo?.inferenceServiceEditInfo,
+  );
+  const [
+    selectedAcceleratorProfile,
+    setSelectedAcceleratorProfile,
+    resetSelectedAcceleratorProfile,
+  ] = useGenericObjectState<AcceleratorProfileSelectFieldState>({
+    profile: undefined,
+    count: 0,
+    useExistingSettings: false,
+  });
+  const customServingRuntimesEnabled = useCustomServingRuntimesEnabled();
+  const [allowCreate] = useAccessReview({
+    ...accessReviewResource,
+    namespace,
+  });
+
+  const [actionInProgress, setActionInProgress] = React.useState(false);
+  const [error, setError] = React.useState<Error | undefined>();
+  const [alertVisible, setAlertVisible] = React.useState(true);
+  const [pvcSize, setPvcSize] = React.useState<string>('');
+  const { dashboardNamespace } = useDashboardNamespace();
+
+  React.useEffect(() => {
+    if (currentProjectName && isOpen) {
+      setCreateDataInferenceService('project', currentProjectName);
+    }
+  }, [currentProjectName, setCreateDataInferenceService, isOpen]);
+
+  // Serving Runtime Validation
+  const isDisabledServingRuntime =
+    namespace === '' || actionInProgress || createDataServingRuntime.imageName === undefined;
+
+  const baseInputValueValid =
+    createDataServingRuntime.numReplicas >= 0 &&
+    resourcesArePositive(createDataServingRuntime.modelSize.resources) &&
+    requestsUnderLimits(createDataServingRuntime.modelSize.resources);
+
+  const isDisabledInferenceService =
+    actionInProgress ||
+    createDataInferenceService.name.trim() === '' ||
+    createDataInferenceService.project === '' ||
+    !isInferenceServiceNameWithinLimit ||
+    !baseInputValueValid;
+
+  const servingRuntimeSelected = React.useMemo(
+    () =>
+      editInfo?.servingRuntimeEditInfo?.servingRuntime ||
+      getServingRuntimeFromName('nvidia-nim-runtime', servingRuntimeTemplates),
+    [editInfo, servingRuntimeTemplates],
+  );
+
+  const onBeforeClose = (submitted: boolean) => {
+    onClose(submitted);
+    setError(undefined);
+    setActionInProgress(false);
+    resetDataServingRuntime();
+    resetDataInferenceService();
+    resetSelectedAcceleratorProfile();
+    setAlertVisible(true);
+  };
+
+  const setErrorModal = (e: Error) => {
+    setError(e);
+    setActionInProgress(false);
+  };
+
+  const onSuccess = () => {
+    setActionInProgress(false);
+    onBeforeClose(true);
+  };
+
+  const submit = () => {
+    setError(undefined);
+    setActionInProgress(true);
+
+    const servingRuntimeName =
+      editInfo?.inferenceServiceEditInfo?.spec.predictor.model?.runtime ||
+      translateDisplayNameForK8s(createDataInferenceService.name);
+
+    const submitServingRuntimeResources = getSubmitServingRuntimeResourcesFn(
```

Comment:
Part of this logic is to flag the project for a use-case -- I'm thinking we need to add yet another param to this function to address the NIM flow... Which I think might be handled by the `NamespaceApplicationCase.KSERVE_PROMOTION` usage -- this would require a backend change to support it to. I'm still thinking on this but FYI. This will make it easier to know if a project is configured for NIM & we can hide things -- like the metrics link (for instance).

Reply:
Any decisions on this one?

Reply:
Please push your other changes -- we can look to address this asap... I need to test something before I can suggest something.